src/app/core/data/root-data.service.ts

```diff
@@ -1,70 +1,33 @@
 import { RequestService } from './request.service';
 import { RemoteDataBuildService } from '../cache/builders/remote-data-build.service';
-import { Store } from '@ngrx/store';
-import { ObjectCacheService } from '../cache/object-cache.service';
 import { HALEndpointService } from '../shared/hal-endpoint.service';
-import { NotificationsService } from '../../shared/notifications/notifications.service';
-import { HttpClient } from '@angular/common/http';
-import { DefaultChangeAnalyzer } from './default-change-analyzer.service';
 import { Observable, of as observableOf } from 'rxjs';
 import { RemoteData } from './remote-data';
 import { FollowLinkConfig } from '../../shared/utils/follow-link-config.model';
-import { PaginatedList } from './paginated-list.model';
-import { CoreState } from '../core-state.model';
-import { FindListOptions } from './find-list-options.model';
 import { DspaceRestService } from '../dspace-rest/dspace-rest.service';
 import { RawRestResponse } from '../dspace-rest/raw-rest-response.model';
 import { catchError, map } from 'rxjs/operators';
-
-
-/**
- * A private DataService implementation to delegate specific methods to.
- */
-class DataServiceImpl extends DataService<Root> {
-  protected linkPath = '';
-  protected responseMsToLive = 6 * 60 * 60 * 1000;
-
-  constructor(
-    protected requestService: RequestService,
-    protected rdbService: RemoteDataBuildService,
-    protected store: Store<CoreState>,
-    protected objectCache: ObjectCacheService,
-    protected halService: HALEndpointService,
-    protected notificationsService: NotificationsService,
-    protected http: HttpClient,
-    protected comparator: DefaultChangeAnalyzer<Root>) {
-    super();
-  }
-}
+import { BaseDataService } from './base/base-data.service';
+import { ObjectCacheService } from '../cache/object-cache.service';
+import { dataService } from './base/data-service.decorator';
 
 /**
  * A service to retrieve the {@link Root} object from the REST API.
  */
 @Injectable()
 @dataService(ROOT)
-export class RootDataService {
-  /**
-   * A private DataService instance to delegate specific methods to.
-   */
-  private dataService: DataServiceImpl;
-
+export class RootDataService extends BaseDataService<Root> {
   constructor(
     protected requestService: RequestService,
     protected rdbService: RemoteDataBuildService,
-    protected store: Store<CoreState>,
     protected objectCache: ObjectCacheService,
     protected halService: HALEndpointService,
-    protected notificationsService: NotificationsService,
-    protected http: HttpClient,
-    protected comparator: DefaultChangeAnalyzer<Root>,
-    protected restService: DspaceRestService) {
-    this.dataService = new DataServiceImpl(requestService, rdbService, null, objectCache, halService, notificationsService, http, comparator);
+    protected restService: DspaceRestService,
+  ) {
+    super('', requestService, rdbService, objectCache, halService, 6 * 60 * 60 * 1000);
```

Comment:
Could we either describe the `6 * 60 * 60 * 1000`, or change it back into a variable (it was previously named `responseMsToLive`)?  It would make this code easier to understand (and easier to replace with a config later if needed).

Reply:
Agreed that putting the `responseMsToLive` in the constructor only is not as readable.
The same applies to `linkPath` and the new `constructIdEndpoint`.

I've changed the constructor signature a bit to move those back into the class body.
"Inner" services still override them in the constructor, since we need to set all of them to the same value for composed services to work properly.

Reply:
As for the "6 hours":
* The root endpoint is not expected to change all that often; we're just caching it for an arbitrarily long while so that we don't spend time refreshing when we really don't need to
* If we make it shorter it should probably still be longer than the default 15 minutes

Reply:
Regarding the constructor refactor ↑
* Looks like the e2e failures were due to the fact that I didn't account for TypeScript here, that refactor would never have worked (but still passed all specs!)
* I've reverted it, but have no other ideas as to how to make this more clear